ethpm/utils/registry.py

```diff
@@ -0,0 +1,87 @@
+import re
+import semver
+
+from urllib import parse
+
+from typing import List
+
+from eth_utils import (
+    is_hex_address,
+)
+
+from ethpm.exceptions import UriNotSupportedError
+
+
+# TODO: update with correct scheme once ERC is completed
+REGISTRY_URI_SCHEME = 'ercxxx'
+PACKAGE_NAME_REGEX = '[a-zA-Z][-_a-zA-Z0-9]{0,255}'
+
+
+def is_valid_authority(auth: str) -> None:
+    """
+    Raises exception if authority is not a valid ENS domain or contract address.
+    """
+    if is_ens_domain(auth) is False and not is_hex_address(auth):
+        raise UriNotSupportedError('{0} is not a valid registry URI authority.'.format(auth))
+
+
+def is_ens_domain(auth: str) -> bool:
+    """
+    Returns false if URI authority is not a valid ENS domain.
+    """
+    if auth[-4:] != '.eth' or len(auth.split('.')) not in [2, 3]:
+        return False
+    return True
+
+
+def is_valid_semver(pkg_version: str) -> None:
+    """
+    Raises exception if package version is not valid semver.
+    """
+    try:
+        semver.parse(pkg_version)
+    except ValueError as exc:
+        raise UriNotSupportedError('{0} is not a valid registry URI version.'.format(pkg_version))
+
+
+def is_valid_package_name(pkg_name: str) -> None:
+    """
+    Returns boolean if the value is a valid package name.
+    """
+    if not bool(re.match(PACKAGE_NAME_REGEX, pkg_name)):
+        raise UriNotSupportedError('{0} is not a valid package name.'.format(pkg_name))
```

Comment:
The naming of the function would make me surprised when this raised an exception as I would expect it to return a boolean.  Maybe move these *validation* functions to a separate module.  I've found it good to have two utils for each of these things.

- `is_valid_thing(value) -> bool` (for *checking*)
- `validate_thing(value) -> None` (either raises or returns `None` for validation)